### PR TITLE
Add isTestRunner utility

### PR DESCRIPTION
**Pull request by yannbf**
### Release notes 

The test-runner adds a `StorybookTestRunner` entry to the browser's user agent. You can use it to determine if a story is rendering in the context of the test runner. This might be useful if you want to disable certain features in your stories when running in the test runner, though it's likely an edge case.

```js
export const MyStory = () => {
  const isTestRunner = window.navigator.userAgent.match(/StorybookTestRunner/)
  return (
    <div>
      <p>Is this story running in the test runner?</p>
      <p>{isTestRunner ? 'Yes' : 'No'}</p>
    </div>
  );
};
```

Given that this check is happening in the browser, it is only applicable in the following scenarios:

- inside of a render/template function of a story
- inside of a play function
- inside of preview.js
- inside any other code that is executed in the browser

<details>
  <summary>📦 Published PR as canary version: <code>0.8.0--canary.198.d45feb9.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/test-runner@0.8.0--canary.198.d45feb9.0
  # or 
  yarn add @storybook/test-runner@0.8.0--canary.198.d45feb9.0
  ```
</details>

### CODE REVIEW

**Review by tmeasday — APPROVED** [review]
Looks great! Thanks @yannbf

File: README.md [hunk 1]
```diff
@@ -521,6 +527,25 @@ module.exports = {
...
+    <p>Is this story running in the test runner?</p>
+    <p>{isTestRunner() ? 'Yes' : 'No'}</p>
+  </div>
+);
+```
+
+The result of `isTestRunner()` will be true in the following scenarios:
+
+1. In the browser, when the story is rendered while running the test runner
+2. In node, if you prepend your Storybook script with `STORYBOOK_TEST_RUNNER=true`
```
Comment (tmeasday) [inline README.md:547]: I'm sort of confused by this usage -- when would someone be using `isTestRunner` in node, and would they have to set the env var themselves?

Comment (yannbf) [reply README.md:547]: If people use the function inside of the play or render functions, or preview.js, it's all good, but if they use it in main.js or to setup a story parameter, the check would not work.

This would be the first step to make this check consistent - now involving setting up the var yourself, but once the test runner is able to spawn storybook (and forward environment variables), then users would not need to do anything and it would be always consistent

Comment (yannbf) [reply README.md]: thought about it more and just removed that bit entirely, it just adds confusion. Once the test-runner is able to spawn Storybook, then I'll bring it back
